### PR TITLE
Expose support for non-returning atomic FADD

### PR DESCRIPTION
**Pull request by mangupta**
- Implements SWDEV-215624

### CODE REVIEW

File: include/hip/hcc_detail/device_library_decls.h [hunk 1]
```diff
@@ -72,6 +72,7 @@ extern "C" __device__ __attribute__((const)) uint __ockl_multi_grid_thread_rank(
 extern "C" __device__ __attribute__((const)) int __ockl_multi_grid_is_valid(void);
 extern "C" __device__ __attribute__((convergent)) void __ockl_multi_grid_sync(void);
 
+extern "C" __device__ void __ockl_global_atomic_add_f32(float*, float);
```
Comment (b-sumner) [inline include/hip/hcc_detail/device_library_decls.h]: The OCKL function requires an address_space(1) pointer for the first argument.

File: include/hip/hcc_detail/hip_atomic.h [hunk 2]
```diff
@@ -73,6 +73,14 @@ float atomicAdd(float* address, float val)
 
     return __uint_as_float(r);
 }
+
+__device__
+inline
+void atomicAddNoRet(float* address, float val)
+{
+    __ockl_global_atomic_add_f32(address, val);
```
Comment (b-sumner) [inline include/hip/hcc_detail/hip_atomic.h]: The OCKL function requires an address_space(1) pointer for the first argument.  Will need a cast if address here.

File: include/hip/hcc_detail/device_library_decls.h (hunk 1 above)
Comment (mangupta) [reply include/hip/hcc_detail/device_library_decls.h]: fixed via b87c508

File: include/hip/hcc_detail/hip_atomic.h (hunk 2 above)
Comment (mangupta) [reply include/hip/hcc_detail/hip_atomic.h]: fixed via b87c508

**Pull request merged**